### ISSUE TITLE
Create search functionality

### ISSUE THREAD

**Issue by omurovec** [privacy-scaling-explorations/pse-gfis#9]
# Overview
Allow users to search the front-end for issues that best suit their skills

**Comment by truthixify** [comment]
I'll do this

**Comment by truthixify** [comment]
@adrianmcli, is results filtering implemented in the backend?
The API only returns JSON object, makes it harder to filter the result.

**Comment by adrianmcli** [comment]
> @adrianmcli, is results filtering implemented in the backend? The API only returns JSON object, makes it harder to filter the result.

Yes, the API essentially spits out a big JSON object. However, it's not so big that the frontend can't filter it pretty quickly. However, the only info in the JSON you can search for is the issue titles.

Happy to discuss how best to implement this search.

---

P.S. To be honest, I think for this feature to be really useful, a list of keywords need to be generated per-repo (on the backend side) as well as per-issue. We can either use RAKE (see #11) or an LLM to do this.

**Comment by truthixify** [comment]
Okay, was thinking of implementing multiple search queries like repo:, title:, author.

I find it really hard to work with the API in dev environment, even CORS won’t allow me to access the API.

**Comment by adrianmcli** [comment]
> Okay, was thinking of implementing multiple search queries like repo:, title:, author.
> 
> 
> 
> I find it really hard to work with the API in dev environment, even CORS won’t allow me to access the API.

Ah yeah, that's unfortunate. I wonder if we can create a test harness so you can easily query GitHub in the dev environment, and work on tweaking the query to get what you need.

If you have a good idea, I'd be happy to merge that. But maybe we should talk about that in a separate issue.

**Comment by truthixify** [comment]
Ok
